fingerprintdb_cuda.cu

```diff
@@ -233,7 +366,14 @@ float FingerprintDB::tanimoto_similarity_cpu(const Fingerprint& fp1,
 size_t get_available_gpu_memory()
 {
     size_t free=0, total=0;
-    cudaMemGetInfo(&free, &total);
+    for(unsigned int i=0; i<get_gpu_count(); i++) {
+        size_t lfree;
+        cudaSetDevice(i);
+        cudaMemGetInfo(&lfree, &total);
+        qDebug() << "GPU" << i << "free:" << lfree;
+        free += lfree;
```

Comment:
I'd extract this since you do it above too:

size_t getDeviceFreeMemory(unsigned int device_index);

Reply:
Done